modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/endofyear/EndOfYearManagerImpl.kt

```diff
@@ -59,48 +71,79 @@ class EndOfYearManagerImpl @Inject constructor(
         hasEpisodesPlayedUpto(YEAR, TimeUnit.MINUTES.toSeconds(EPISODE_MINIMUM_PLAYED_TIME_IN_MIN)) &&
             FeatureFlag.isEnabled(Feature.END_OF_YEAR_ENABLED)
 
-    /**
-     * Download the year's listening history.
-     */
+    private var synced: Double = 0.0
+    private var syncTotal: Double = 0.0
+
     override suspend fun downloadListeningHistory(onProgressChanged: (Float) -> Unit) {
         if (!syncManager.isLoggedIn()) {
             return
         }
+        resetSyncCount()
+        coroutineScope {
+            awaitAll(
+                *yearsToSync.map { year ->
+                    // Download listening history for each year in parallel
+                    if (BuildConfig.DEBUG) {
+                        Timber.i("End of Year: Downloading listening history for year $year")
+                    }
+                    async { downloadListeningHistory(year, onProgressChanged) }
+                }.toTypedArray()
+            )
+        }
+        if (BuildConfig.DEBUG) {
+            Timber.i("End of Year: listening history sync complete")
+        }
+        onProgressChanged(1f)
+    }
+
+    /**
+     * Download the year's listening history.
+     */
+    private suspend fun downloadListeningHistory(year: Int, onProgressChanged: (Float) -> Unit) {
         // check for an episode interacted with before this year and assume they have the full listening history if they exist
-        if (anyEpisodeInteractionBeforeYear()) {
+        if (anyEpisodeInteractionBeforeYear(year)) {
             return
         }
         // only download the count to check if we are missing history episodes
-        val countResponse = syncManager.historyYear(year = YEAR, count = true)
+        val countResponse = syncManager.historyYear(year = year, count = true)
         onProgressChanged(0.1f)
         val serverCount = countResponse.count ?: 0
-        val localCount = countEpisodeInteractionsInYear()
+        val localCount = countEpisodeInteractionsInYear(year)
         Timber.i("End of Year: Server listening history. server: ${countResponse.count} local: $localCount")
         if (serverCount > localCount) {
             // sync the year's listening history
-            val response = syncManager.historyYear(year = YEAR, count = false)
+            val response = syncManager.historyYear(year = year, count = false)
             onProgressChanged(0.2f)
             val history = response.history ?: return
             historyManager.processServerResponse(
                 response = history,
                 updateServerModified = false,
+                updateSyncTotal = { syncTotal += it },
                 onProgressChanged = {
-                    onProgressChanged(0.2f + (it * 0.8f))
+                    synced += 1
+                    val progress = min((0.2f + (synced / syncTotal) * 0.8f), 0.95).toFloat()
```

Comment:
Just double checking, but this `min` call means that the number will never get over `0.95`, right? It might be worth a comment explaining why we want that.

Reply:
Added a comment here: https://github.com/Automattic/pocket-casts-android/pull/1522/commits/e5d3b1e0b921b520556b300aabb2d06a3d518ebc#diff-d4ead7d8903c1b9b64124a86579385d509abee41144951e6ed7f0a2b1f32a2efR78-R80